miranda/hq/daily.py

```diff
@@ -0,0 +1,254 @@
+        txt = fh.read()
+        meta, data = re.split(data_header_pattern, txt, maxsplit=2)
+
+    sections = iter(re.split(section_patterns, meta)[1:])
+
+    m = dict()
+    for sec in sections:
+        if sec in meta_patterns:
+            content = next(sections)
+            for key, pat in meta_patterns[sec].items():
+                match = re.search(pat, content)
+                if match:
+                    m[key] = match.groups()[0]
+
+        elif sec in csv_patterns:
+            content = next(sections).strip()
+            m[sec] = list(
+                csv.reader(content.splitlines(), dialect="unix", delimiter=";")
+            )
+
+    d = pd.read_csv(
+        path,
+        delimiter=";",
+        skiprows=len(meta.splitlines()),
+        encoding="latin1",
+        index_col=0,
+        parse_dates=True,
+        infer_datetime_format=True,
+        decimal=",",
+    )
+
+    return m, d
+
+
+def to_cf(
+    meta: dict, data: pd.DataFrame, cf_table: Optional[dict] = None
+) -> xr.DataArray:
+    """Return CF-compliant metadata."""
+
+    if dict is None:
+        cf_table = cmor
+
+    # Convert meta values
+    m = dict()
+    for key, val in meta.items():
+        m[key] = converters.get(key, lambda x: x)(val)
+
+    # Get default variable attributes
+    name = guess_variable(m, cf_table)
+    attrs = cf_table[name]
+
+    # Add custom HQ attributes
+    for key, val in cf_attrs_names.items():
+        if key in m:
+            attrs[val] = m[key]
+
+    x = data["Valeur"].values.astype(float)
+
+    # Convert units
+    if attrs["units"] != m["unité"]:
+        x = unites.convert(x, m["unité"], units2pint(attrs["units"]))
+
+    coords = {k: attrs.pop(k, np.nan) for k in ["lon", "lat", "elevation", "site"]}
+    coords["time"] = data.index.values
+    cf_corrected = xr.DataArray(
+        data=x, dims="time", coords=coords, name=name, attrs=attrs
+    )
+    return cf_corrected
+
+
+def open_csv(path: Union[str, Path], cf_table: Optional[dict] = None) -> xr.DataArray:
+    """Extract daily HQ meteo data and convert to xr.DataArray with CF-Convention attributes."""
+    meta, data = extract_daily(path)
+    return to_cf(meta, data, cf_table)
+
+
+def units2pint(value: str) -> pint.Quantity:
+    """Return the pint Unit for the DataArray units.
+
+    Parameters
+    ----------
+    value : str
+      Unit expression.
+
+    Returns
+    -------
+    pint.Quantity
+      Pint compatible units.
+
+    """
+
+    def _transform(s):
+        """Convert a CF-unit string to a pint expression."""
+        return re.subn(r"\^?(-?\d)", r"**\g<1>", s)[0]
+
+    value = value.replace("%", "pct")
+    try:  # Pint compatible
+        return unites.parse_expression(value).units
+    except (
+        pint.UndefinedUnitError,
+        pint.DimensionalityError,
+    ):  # Convert from CF-units to pint-compatible
+        return unites.parse_expression(_transform(value)).units
+
+
+# if __name__ == "__main__":
```

Comment:
Maybe just remove this, verify that everything still works with the small changes I made and merge away.